editor/store/reducer.js

```diff
@@ -114,7 +169,26 @@ export const editor = flow( [
 	combineReducers,
 
 	// Track undo history, starting at editor initialization.
-	partialRight( withHistory, { resetTypes: [ 'SETUP_NEW_POST', 'SETUP_EDITOR' ] } ),
+	partialRight( withHistory, {
+		resetTypes: [ 'SETUP_NEW_POST', 'SETUP_EDITOR' ],
+		shouldOverwriteState: ( previous, next ) => {
+			if ( previous.type !== next.type || previous.type !== 'UPDATE_BLOCK_ATTRIBUTES' ) {
+				return false;
+			}
+
+			const previousAttributes = keys( previous.attributes );
+			const nextAttributes = keys( next.attributes );
+			if (
+				previousAttributes.length !== 1 ||
+				nextAttributes.length !== 1 ||
+				previousAttributes[ 0 ] !== nextAttributes[ 0 ]
+			) {
+				return false;
+			}
+
+			return areValuesEquivalent( previous.attributes[ previousAttributes[ 0 ] ], next.attributes[ nextAttributes[ 0 ] ] );
```

Comment:
`areValuesEquivalent` sounds like an expensive calculation to execute on every action (an thus keystroke).